src/shared/components/Banners/Banners.tsx

```diff
@@ -1,21 +1,32 @@
 import { Collapse } from '@mui/material';
 import { TransitionGroup } from 'react-transition-group';
+import { useCallback } from 'react';
 
-import { BannerComponents, banners } from 'shared/state/Banners';
+import { BannerPayload, banners } from 'shared/state/Banners';
 import { useAppDispatch } from 'redux/store';
 
+import { BannerComponents } from './Banners.const';
+
 export const Banners = () => {
   const dispatch = useAppDispatch();
   const data = banners.useData();
 
+  const handleClose = useCallback(
```

Comment:
Would it be beneficial to use useCallback here? What advantages would this decision provide?

Reply:
Per my understanding, to not rerender `<BannerComponent ... />` which uses it this property. 

But, in this case, to prevent redundant renders of all `<BannerComponent...` from the list, it should have a bit another location and sequence of contexts for the property `handleClose`:

1). why we need to think about `useCallback` when we don't have a huge calculations/bad performance in descendants or in current component? I suppose it looks like overengineering here and in other similar places.

2.1). it should be outside of the component `Banners` to avoid the generating of new reference of the function each time when `const data = banners.useData();` is updated
```
const handlePureClose = (dispatch: AppDispatch, { key, bannerProps }: BannerPayload) => () => {
  dispatch(banners.actions.removeBanner({ key }));
  bannerProps?.onClose?.();
}
```
2.2). not to generate new anonymous function when banners data is updated:
```
<BannerComponent {...bannerProps} onClose={handlePureClose({ key, bannerProps })} />
```